poller.go

```diff
@@ -254,6 +254,9 @@ func (p *Poller) Wait(ctx context.Context, interval time.Duration) <-chan error
 									successful = 0
 									continue
 								}
+
+								NotifyMetrics(fmt.Sprintf("initial value %g, current value %g (%s elapsed)", initVal, resVal, p.timer.Elapsed()))
```

Comment:
You should make sure `NotifyMetrics` isn't nil here before calling it. Also a little odd it is capitalized.

Reply:
Both good points. Addressed cb26548